.github/workflows/build-artifact-upload.yml

```diff
@@ -0,0 +1,56 @@
+name: Build Artifact Upload
+
+on:
+  push:
+    branches:
+      - master
```

Comment:
What do you think about changing this to 'main' while we're touching it? Or would that break other things?